plugins/inputs/icecast/icecast.go

```diff
@@ -0,0 +1,197 @@
+package icecast
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// SourceListmounts holds single listener elements from the Icecast XML
+type SourceListmounts struct {
+	Mount       string `xml:"mount,attr"`
+	Listeners   int32  `xml:"listeners"`
+	Connected   int32  `xml:"connected"`
+	ContentType string `xml:"content-type"`
+}
+
+// Listmounts main structure of the icecast XML
+type Listmounts struct {
+	Sources []SourceListmounts `xml:"source"`
+}
+
+// Icecast contains all the required details to connect to
+type Icecast struct {
+	Servers         map[string]server `toml:"servers"`
+	ResponseTimeout internal.Duration `toml:"response_timeout"`
+	Slash           bool              `toml:"slash"`
+}
+
+type server struct {
+	URL      string
+	Alias    string
+	Username string
+	Password string
+}
+
+var sampleConfig = `
+  ## Specify the IP adress/hostname to where the '/admin/listmounts' can be found. You can include port if needed.
+  ## You can also specify an alias. If none is given, the hostname will be used. Multiple servers can also be specified
+  [servers]
+	[server.1]
+	url = "http://localhost"
+	alias = "Server 1"
+	username = "telegraf"
+	password = "passwd"	
+
+  ## Timeout to the complete conection and reponse time in seconds. Default (5 seconds)
+  # response_timeout = "25s"
+
+  ## Include the slash in mountpoint names or not
+  slash = false
+`
+
+// SampleConfig is called upon when auto-creating a configuration
+func (n *Icecast) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description is used to describe the input module
+func (n *Icecast) Description() string {
+	return "Read listeners from an Icecast instance per mount"
+}
+
+// Gather will fetch the metrics from Icecast
+func (n *Icecast) Gather(acc telegraf.Accumulator) error {
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	var outerr error
+	var errch = make(chan error)
+
+	for _, s := range n.Servers {
+		server := s
+
+		// Default admin listmounts page of Icecast
+		adminPageURL := "/admin/listmounts"
+
+		// Parsing the URL to see if it's ok
+		addr, err := url.Parse(server.URL)
+		if err != nil {
+			return fmt.Errorf("Unable to parse address '%s': %s", server.URL, err)
+		}
+		addr.Path = path.Join(addr.Path, adminPageURL)
+		go func(addr *url.URL) {
+			errch <- n.gatherURL(addr, server, acc)
+		}(addr)
+	}
+
+	// Drain channel, waiting for all requests to finish and save last error.
+	for range n.Servers {
+		if err := <-errch; err != nil {
+			outerr = err
+		}
+	}
```

Comment:
This supports my feeling that it would be much easier to only handle a single server with this plugin and use the possibility to add this plugin multiple times to the config. This way the code will simplify a lot!